frontend/src/ui/Access.tsx

```diff
@@ -643,21 +648,10 @@ interface GroupDag {
     sort(groups: Option[]): Option[];
 }
 
-const groupDAG: () => GroupDag = (() => {
-    let instance: GroupDag | null = null;
-    return () => {
-        if (!instance) {
-            instance = buildDag();
-        }
-
-        return instance;
-    };
-})();
-
-const buildDag = (): GroupDag => {
+const buildDag = (knownGroups: KnownRoles): GroupDag => {
```

Comment:
This doesn't use the supersets/`implies` entries from `groups.json` yet (there is a `TODO` for that in line 662). Do you plan to do this in a follow up PR?